scripts/run_full_validation.sh

```
@@ -0,0 +1,64 @@
+echo "Simulating merge with origin/main..."
+git merge --no-commit --no-ff origin/main >/dev/null 2>&1 || {
+  echo "Merge conflicts detected. Resolve before running validation." >&2
+  exit 1
+}
+
+# Run all configured pre-commit hooks
+if git rev-parse --show-toplevel >/dev/null 2>&1; then
+  echo "Running pre-commit hooks across the repository..."
+  pre-commit run --all-files
+  echo "Running pre-push pytest suite..."
+  SKIP= pytest pre-commit run pytest --hook-stage pre-push
```

Shellcheck (0.11.0)

[warning] Remove space after = if trying to assign a value (for empty string, use var='' ... ).

(SC1007)